examples/example_composed_target_function.py

```diff
@@ -0,0 +1,160 @@
+    """Objective function of the regression task.
+
+    Parameters
+    ----------
+    individual : Individual
+        Individual of the Cartesian Genetic Programming Framework.
+    target_function : Callable
+        Target function.
+    rng: numpy.random.RandomState
+
+    Returns
+    -------
+    Individual
+        Modified individual with updated fitness value.
+    """
+    if individual.fitness is not None:
+        return individual
+
+    n_function_evaluations = 1000
+
+    f_graph = individual.to_func()
+    y = np.empty(n_function_evaluations)
+    x = rng.uniform(-5, 5, size=n_function_evaluations)
+    for i, x_i in enumerate(x):
+        y[i] = f_graph([x_i])[0]
+
+    loss = np.mean((target_function(x) - y) ** 2)
+    individual.fitness = -loss
+
+    return individual
+
+
+def evolution(target_function, rng):
+    """Execute CGP on a regression task for a target function composed of two sub-functions.
+
+    Parameters
+    ----------
+    target_function : Callable
+        Target function
+    rng: numpy.random.RandomState
+
+    Returns
+    -------
+    dict
+        Dictionary containing the history of the evolution
+        with parents fitness, dna of fittest individual in every generation
+    Individual
+        Individual with the highest fitness in the last generation
+    """
+    population_params = {"n_parents": 1, "mutation_rate": 0.005, "seed": 8188211}
+
+    genome_params = {
+        "n_inputs": 1,
+        "n_outputs": 1,
+        "n_columns": 20,
+        "n_rows": 1,
+        "levels_back": None,
+        "primitives": (
+            ComparisonOperator,
+            IfElseOperator,
+            cgp.Mul,  # needed for x² if x>0
+            ConstantFloatZero,  # needed for comparison operator
+        ),
+    }
+
+    ea_params = {"n_offsprings": 4, "tournament_size": 2, "n_processes": 2}
+
+    evolve_params = {"max_generations": 1000, "min_fitness": 0.0}
+
+    # create population that will be evolved
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+
+    # create instance of evolutionary algorithm
+    ea = cgp.ea.MuPlusLambda(**ea_params)
+
+    # define callback for recording of fitness over generations
+    history = {}
+    history["fitness_parents"] = []
+    history["champion_dna"] = []
+
+    def recording_callback(pop):
+        history["fitness_parents"].append(pop.fitness_parents())
+        history["champion_dna"].append(pop.champion.genome.dna)
+
+    # the objective passed to evolve should only accept one argument,
+    # the individual
+    obj = functools.partial(objective, target_function=target_function, rng=rng)
+
+    # Perform the evolution
+    cgp.evolve(pop, obj, ea, **evolve_params, print_progress=True, callback=recording_callback)
+    return history, pop.champion
+
+
+if __name__ == "__main__":
+    seed = 1234  # Todo: find better seed
+    rng = np.random.RandomState(seed=seed)
+
+    history, champion = evolution(f_target, rng)
```

Comment:
Can you add a plot here, which can then be rendered in the documentation?
And can you also print the champion's expression in the end?